apiserver/storage/storage.go

```diff
@@ -756,3 +756,45 @@ func (a *API) Destroy(args params.Entities) (params.ErrorResults, error) {
 	}
 	return params.ErrorResults{result}, nil
 }
+
+// Attach attaches existing storage instances to units.
+// A "CHANGE" block can block this operation.
+func (a *API) Attach(args params.StorageAttachParams) (params.ErrorResults, error) {
+	if err := a.checkCanWrite(); err != nil {
+		return params.ErrorResults{}, errors.Trace(err)
+	}
+
+	// Check if changes are allowed and the operation may proceed.
+	blockChecker := common.NewBlockChecker(a.storage)
+	if err := blockChecker.ChangeAllowed(); err != nil {
+		return params.ErrorResults{}, errors.Trace(err)
+	}
+
+	result := make([]params.ErrorResult, len(args.Attachments))
+	for i, arg := range args.Attachments {
```

Comment:
To avoid churn later, break out into a separate attachOne() func now in this PR?